restconf_update_ipaddress/updateip.py

```diff
@@ -26,49 +27,46 @@
 import sys
 from argparse import ArgumentParser
 from collections import OrderedDict
+from getpass import getpass
 import urllib3
 
 # Disable SSL Warnings
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-
-# These variables target the RESTCONF Always-On Sandbox hosted by Cisco DevNet
-HOST = 'ios-xe-mgmt.cisco.com'
-PORT = '443'
-USER = 'root'
-PASS = 'D_Vay!_10&'
-
 # Identifies the interface on the device used for management access
 # Used to ensure the script isn't used to update the IP leveraged to manage device
 MANAGEMENT_INTERFACE = "GigabitEthernet1"
 
-# Create the base URL for RESTCONF calls
-url_base = "https://{h}:{p}/restconf".format(h=HOST, p=PORT)
-
 # Identify yang+json as the data formats
 headers = {'Content-Type': 'application/yang-data+json',
            'Accept': 'application/yang-data+json'}
 
 
 # Function to retrieve the list of interfaces on a device
-def get_configured_interfaces():
-    url = url_base + "/data/ietf-interfaces:interfaces"
-
+def get_configured_interfaces(url_base, username, password):
     # this statement performs a GET on the specified url
-    response = requests.get(url,
-                            auth=(USER, PASS),
-                            headers=headers,
-                            verify=False
-                            )
+    try:
+        response = requests.get(url_base,
+                                auth=(username, password),
+                                headers=headers,
+                                verify=False
+                                )
+    except Exception as e:
+        print(e, file=sys.stderr)
+        sys.exit(1)
+
+    if response.status_code >= 300:
```

Comment:
You don't need this if you call `raise_for_status()`.  Any HTTP status code that is an error will turn into an exception.

Reply:
done